File: cmcc_monitor/src/main/scala/com/sheep/cmcc/utils/Jpools.scala
package cmcc_monitor.src.main.scala.com.sheep.cmcc.utils

import org.apache.commons.pool2.impl.GenericObjectPoolConfig
import redis.clients.jedis.{Jedis, JedisPool}

object Jpools {
  private val poolConfig = new GenericObjectPoolConfig()
  //连接池中最大的空闲连接数，默认是8
  poolConfig.setMaxIdle(5)
  //只支持最大的连接数，连接池中最大的连接数，默认是8
  poolConfig.setMaxTotal(2000)
  private lazy val jedisPool: JedisPool = new JedisPool(poolConfig,AppParams.redisHost)

  def getJedis = {
    val jedis: Jedis = jedisPool.getResource
    jedis.select(AppParams.redisDbIndex)
    jedis
  }
}

File: cmcc_visual_36/src/main/java/cn/sheep/cmcc/utils/Jpools.java
package cmcc_visual_36.src.main.java.cn.sheep.cmcc.utils;

import com.typesafe.config.Config;
import com.typesafe.config.ConfigFactory;
import org.apache.commons.pool2.impl.GenericObjectPoolConfig;
import redis.clients.jedis.Jedis;
import redis.clients.jedis.JedisPool;

/**
 * redis数据库连接池访问类
 */
public class Jpools {
    private static Config load = ConfigFactory.load();
    private static GenericObjectPoolConfig poolConfig = new GenericObjectPoolConfig();
    private static JedisPool jedisPool = null;
    static {
        poolConfig.setMaxIdle(load.getInt("redis.pool.maxActive"));
        poolConfig.setMaxIdle(load.getInt("redis.pool.maxIdle"));
       jedisPool =  new JedisPool(poolConfig,load.getString("redis.ip"),load.getInt("redis.port"));
    }
    //提供redis访问连接
    public static Jedis getJedis(){
        Jedis jedis = jedisPool.getResource();
        jedis.select(load.getInt("redis.db.index"));
        return jedis;
    }
}
